### PR TITLE
[reg] Add reg chart

### PR DESCRIPTION
**Pull request by nicholaswilde**
Signed-off-by: Nicholas Wilde <ncwilde43@gmail.com>

**Description of the change**

Adds `reg` chart

**Benefits**

New chart.

**Possible drawbacks**

None

**Applicable issues**

None

**Additional information**

`env.EXTRA_ARGS` is commented out in `values.yaml` in order to give direction on how to start the reg server.

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [X] Variables are documented in the README.md (this can be done with using our helm-docs wrapper `./hack/gen-helm-docs.sh stable <chart>`)

### CODE REVIEW

File: charts/stable/reg/README_CHANGELOG.md.gotmpl [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+- N/A
+
+#### Changed
+
+- N/A
+
+#### Removed
+
+- N/A
```
Comment (bjw-s) [inline charts/stable/reg/README_CHANGELOG.md.gotmpl:24]:
```suggestion
#### Added

- Initial version
```

File: charts/stable/reg/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+apiVersion: v2
+appVersion: 0.16.1
+description: Docker registry v2 command line client and repo listing generator with security checks.
+name: reg
+version: 1.0.0
+kubeVersion: ">=1.16.0-0"
+keywords:
+- reg
+home: https://github.com/k8s-at-home/charts/tree/master/charts/stable/reg
+sources:
```
Comment (bjw-s) [inline charts/stable/reg/Chart.yaml:11]: It's normally a good practice to also include some sort of icon. Maybe there is some sort of Github avatar for `genuinetools` that can be "borrowed"?

File: charts/stable/reg/values.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  pullPolicy: IfNotPresent
+  tag: v0.16.1
+
+strategy:
+  type: Recreate
+
+# See more environment variables in the reg documentation
+# https://github.com/genuinetools/reg
+env:
+  EXTRA_ARGS: "server --registry r.j3ss.co"
```
Comment (bjw-s) [inline charts/stable/reg/values.yaml:19]: I'm not familiar with the app, but is this a good value to have as default?

**Review by bjw-s — CHANGES_REQUESTED** [review]

File: charts/stable/reg/Chart.yaml (hunk 2 above)
Comment (nicholaswilde) [reply charts/stable/reg/Chart.yaml:11]: I can take a look. I know that there isn't one for the app.

File: charts/stable/reg/values.yaml (hunk 3 above)
Comment (nicholaswilde) [reply charts/stable/reg/values.yaml:19]: The ci tests fail if we don't specify a server to run.

Comment (bjw-s) [reply charts/stable/reg/values.yaml:19]: Ah, in that case you can create a `ci/ct-values.yaml` file containing this. That way the CI will use that and we can remove the default here.

Comment (nicholaswilde) [reply charts/stable/reg/values.yaml:19]: @bjw-s I'm trying to determine the best route to implement `ci/ct-values.yaml`. I'm looking at `charts-lint-test.yaml` and trying to see if I should add on to the `changes-install` and `install` jobs in `charts-lint-test.yaml`, or if I should just create a whole new job. Which would you recommend?

Comment (onedr0p) [reply charts/stable/reg/values.yaml:19]: You just need to place values in `ci/ct-values.yaml` that you want to run only on CI

**Review by bjw-s — APPROVED** [review]
I took the liberty of adding the ci/ct-values.yaml file, because the chart looks good otherwise. Merging! 🎉

**Comment by nicholaswilde** [comment]
> I took the liberty of adding the ci/ct-values.yaml file, because the chart looks good otherwise. Merging! 🎉 

I appreciate it! Thanks for the help!

**Pull request merged**